### PR TITLE
Confirm close on create post

### PR DESCRIPTION
**Pull request by antonetteortiz**
[#974 Confirm close on Create Modal ](https://github.com/Hylozoic/hylo-evo/issues/974)

Description
--
If user makes any changes to the Create Modal and clicks outside of the window or closes the modal, a confirmation prompt will appear. 

How to Test:
---
- In the create discussion modal, make changes to any field.
- Click outside of the window and/or attempt to close the modal.
- Confirmation window will popup, prompting to either return to the create modal or discard post.

### CODE REVIEW

File: src/components/CreateModal/CreateModal.connector.js [hunk 1]
```diff
@@ -15,9 +15,15 @@ export const mapDispatchToProps = (dispatch, props) => {
   const closeUrl = postId
     ? postUrl(postId, urlParams, querystringParams)
     : addQuerystringToPath(baseUrl(urlParams), querystringParams)
+  const closeModal = () => dispatch(push(closeUrl))
 
   return {
-    closeModal: () => dispatch(push(closeUrl))
+    closeModal,
+    confirmLeaveCreateModal: () => {
```
Comment (tibetsprague) [inline src/components/CreateModal/CreateModal.connector.js:22]: this works great, but could be slightly less code and cleaner by just passing an isDirty parameter into closeModal, then you wouldn't need two functions

File: src/components/CreateModal/CreateModal.js [hunk 2]
```diff
@@ -8,30 +8,62 @@ import PostEditor from 'components/PostEditor'
...
-        <Switch>
-          <Route path={match.path + `/post`} children={({ match, location }) =>
-            <PostEditor match={match} location={location} onClose={closeModal} />} />
-          <Route path={match.path + `/group`} children={({ match, location }) =>
-            <CreateGroup match={match} location={location} onClose={closeModal} />} />
-          <Route>
-            <CreateModalChooser match={match} location={location} />
-          </Route>
-        </Switch>
+  const checkBeforeClose = () => {
```
Comment (tibetsprague) [inline src/components/CreateModal/CreateModal.js:16]: if you pass in isDirty to closeModal then this conditional moves to closeModal

File: src/components/PostEditor/PostEditor.js [hunk 3]
```diff
@@ -106,7 +106,8 @@ export default class PostEditor extends React.Component {
       toggleAnnouncementModal: false,
       showPostTypeMenu: false,
       titleLengthError: false,
-      dateError: false
+      dateError: false,
+      isDirty // prop tracking if form fields were touched from CreateModal
```
Comment (tibetsprague) [inline src/components/PostEditor/PostEditor.js:110]: it doesnt look like you are actually using this, why do you need to pass isDirty down to the PostEditor?

File: src/components/PostEditor/PostEditor.js [hunk 4]
```diff
@@ -118,6 +119,7 @@ export default class PostEditor extends React.Component {
     this.editor = React.createRef()
     this.groupsSelector = React.createRef()
     this.topicSelector = React.createRef()
+    this.setIsDirty = props.setIsDirty
```
Comment (tibetsprague) [inline src/components/PostEditor/PostEditor.js:122]: you dont need to do this, you can just call this.props.setIsDirty down below

File: src/components/PostEditor/PostEditor.js [hunk 5]
```diff
@@ -278,6 +284,8 @@ export default class PostEditor extends React.Component {
       this.setState({
         detailsTopics: topicNames.map(tn => ({ label: tn, name: tn, id: tn }))
       })
+      // TODO (Annie): figure out why topic changes don't update isDirty
```
Comment (tibetsprague) [inline src/components/PostEditor/PostEditor.js:287]: hmm, strange, we can look at this together

**Review by tibetsprague — CHANGES_REQUESTED** [review]
whoo hoo, looking good! just a couple little suggestions/changes requested, and we can look at the topics thing together

**Review by tibetsprague — COMMENTED** [review]
oh also we will want to add/tweak our tests to cover the changes, so good opportunity to look at our test code together too

**Comment by antonetteortiz** [comment]
thanks Tibet, I will work on the change request and connect with you later today to look at topic changes.

**Review by tibetsprague — APPROVED** [review]
i see you went ahead and cleaned up a lot of code, thanks :)

**Pull request merged**